File: 209. Minimum Size Subarray Sum/C++/Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int> &nums) {
        int i = 0;
        int theSum = 0;
        int theMin = INT_MAX;
        for (int j = 0; j < nums.size(); j++) {
            theSum += nums[j];
            while (theSum >= s) {
                theMin = min(theMin, j - i + 1);
                if (theMin == 1) {
                    return theMin;
                }
                theSum -= nums[i];
                i++;
            }
        }
        return theMin == INT_MAX ? 0 : theMin;
    }
};

File: 209. Minimum Size Subarray Sum/Python/Minimum Size Subarray Sum.py
class Solution(object):
    def minSubArrayLen(self, s, nums):
        """
        :type s: int
        :type nums: List[int]
        :rtype: int
        """
        i = 0
        the_sum = 0
        the_min = sys.maxsize
        for j, num in enumerate(nums):
            the_sum += num
            while the_sum >= s:
                the_min = min(the_min, j - i + 1)
                if the_min == 1:
                    return the_min
                the_sum -= nums[i]
                i += 1
        return 0 if the_min == sys.maxsize else the_min
